### DISCUSSION TITLE
What are the core services?

### DISCUSSION THREAD

**Discussion by Solarunit** [immich-app/immich#3020] [Q&A] [answered]
I am interesting to try immich, and wonder what are the core services in the compose file? There are a lot of them and it's confusing.

- immich-server
- immich-microservices
- immich-machine-learning
- immich-web
- typesense
- redis
- database
- immich-proxy

What services I can remove from the compose file?
I assume these ones are not essential:

- immich-microservices
- immich-machine-learning
- typesense

**Comment by jrasm91** [comment, marked as answer]
Immich web is the webserver for the web app, it is essential in order to use the web interface

Immich server is the main API, it is essential

Immich microservices is the background process for all the jobs, including exif parsing, it is essential

Redis is how the server and microservices communicate, it is essential

Postgres is obviously essential, it is where all the data is persisted

Typesense is optional. It provides search functionality and supports the facial recognition features.

Machine learning is optional.

Immich proxy is optional, but you will need to configure your own reverse proxy for the web and server to communicate properly.

**Reply by Solarunit** [reply]
Thanks for the detailed reply!

**Comment by Solarunit** [comment]
Looks like it can't run without Typesense, I got 500 error without it.

**Reply by jrasm91** [reply]
TYPESENSE_ENABLED=false

**Reply by jrasm91** [reply]
https://immich.app/docs/install/environment-variables#typesense

**Reply by Solarunit** [reply]
yes, I think I did "TYPESENSE_ENABLED=false", and still got 500 error

**Reply by Solarunit** [reply]
oh, I am running immich now btw, only immich-machine-learning is commented in the compose file
